astr303a.py

```diff
@@ -0,0 +1,571 @@
+
+    #Plot galaxies and cluster center
+    plt.title('Figure 3: Coma Cluster Galaxies',fontsize=14)
+    plt.ylabel('Declination '+'$ra\,(degrees)$',fontsize=12)
+    plt.xlabel('Right Ascension '+r'$dec\,(degrees)$',fontsize=12)
+    plt.xlim(193.5,196.5)
+    plt.ylim(26.5,29.5)
+    for i in range(0,len(dataarray2)):
+        plt.plot(dataarray2[i][1],dataarray2[i][2],markeredgecolor='black',marker='o',markerfacecolor='None',markersize=3)
+    for i in range(0,len(dataarray)):
+        plt.plot(dataarray[i][1],dataarray[i][2],color='black',marker='o',markersize=3,fillstyle=None)
+    plt.plot(comaRA,comaDEC,color='red',marker='*',markersize=14,label='Center',linestyle='None')
+    plt.plot(0,0,color='black',marker='o',markersize=3,fillstyle=None,label='Member',linestyle='None')
+    plt.plot(0,0,markeredgecolor='black',marker='o',markerfacecolor='None',markersize=3,label='Non-member',linestyle='None')
+    
+    #Plot ellipse
+    ratio = 1/np.cos(np.pi*comaDEC/180) #compute major axis distortion at declination of cluster center
+    ellipseA = ((1.5*180)/(np.pi*dcoma))*ratio #set major axis
+    ellipse = Ellipse((comaRA,comaDEC),width=2*(ellipseA),height=2*(1.5*180)/(dcoma*np.pi),linestyle=':',fill=False)
+    fig = plt.gcf()
+    ax = fig.gca()
+    plt.legend(loc='upper center',ncol=3,fontsize=10,numpoints=1)
+    plt.gca().invert_xaxis()
+    ax.add_patch(ellipse)
+    plt.grid()
+    pp.savefig()
+    plt.clf()
+
+    #create list of all redshifts
+    redshift = []
+    redshift2 = []
+    for i in range(0,len(dataarray)):
+        redshift.append(dataarray[i][6])
+        redshift2.append(dataarray[i][6])
+    for i in range(0,len(dataarray2)):
+        redshift2.append(dataarray2[i][6])
+
+    #Gather data in bins of width 0.001. Retrive counts from each bin.
+    histogram = py.hist(redshift,bins=np.arange(0.005, 0.05, 0.001),color='lightgrey')
+    counts = histogram[0] #get counts
+    shifts = histogram[1] #get z values of bins
+    binset = []
+
+    #Create list of z values at bin centers
+    for i in range(0,len(shifts)-1):
+        binset.append(shifts[i]+0.0005)
+
+    #Define Gaussian, calculate initial parameter guesses.
+    def gauss(x, a, x0, sigma):
+        return a*np.exp(-(x-x0)**2/(2*sigma**2))
+    stddev = np.std(redshift)
+    mean = np.mean(redshift)
+
+    #Fit gaussian
+    coeffs, pcov = optimize.curve_fit(gauss, binset, counts, p0 = [1, mean, stddev])       
+
+    #Plot histogram and fitted Gaussian.
+    plt.title('Figure 4: Coma Cluster Member Galaxy Redshifts')
+    plt.xlabel('Redshift '+r'$z$', fontsize = 12)
+    plt.ylabel('Number of galaxies', fontsize = 12)
+    plt.xlim(0,0.05)
+    plt.plot(binset, gauss(binset, *coeffs),color='black',linestyle='-')
+    plt.annotate(r'$\bar z$'+' ='+str('{:.4E}'.format(comameanz))+'\n'+r'$\sigma_z$'+' ='+str('{:.4E}'.format(abs(coeffs[2]))), xy=(0.70, 0.90), xycoords='axes fraction',fontsize = 10)
+    pp.savefig()
+    plt.clf()
+    
+    return
```

Comment:
This is unnecessary. If you're not returning anything, then running out the end of the function will do exactly this. But if the output of this function is a plot, you could consider better off returning the plot object.